### PR TITLE
feat client sdk -Add publication module

### PR DESCRIPTION
**Pull request by paulxuca**

### CODE REVIEW

File: packages/client/src/publication/index.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+export * from './Publication';
+export {
+  PublicationMetadataDisplayTypes,
+  PublicationContentWarning,
+  PublicationMainFocus,
+} from '../graphql/types.generated';
```
Comment (paulxuca) [inline packages/client/src/publication/index.ts:6]: @krzysu Not sure which enums I should export for this module -- let me know if there's a good spot to find this out!

**Comment by krzysu** [comment]
there are two possible methods missing from this PR, could you also update TODO file? please keep "method descriptor" (`[PQ]`) as it will help us later when adding pagination or `execute` helpers.

```
- [Q] client.publication.fetch(request)
- [PQ] client.publication.fetchAll(request)
- [Q] client.publication.validateMetadata(request)
- [PQ] client.publication.whoCollected({ publicationId })

- [PQ] client.publication.allForSale({ profileId })
- [Q] client.publication.metadataStatus(request)
```

File: packages/client/src/publication/index.ts (hunk 1 above)
Comment (krzysu) [reply packages/client/src/publication/index.ts:6]: I guess you just have to check what the `request` argument offers for different publication queries and make it all available

File: packages/client/TODO.md [hunk 2]
```diff
@@ -93,13 +93,12 @@ Client manages the token lifecycle and storage
...
-- [PQ] client.publication.whoCollected({ publicationId })
-
-- [PQ] client.publication(s).forSale({ profileId })
-- [Q] client.publication.metadataStatus(request)
+- [x] [Q] client.publication.fetch(request) or client.publications.fetchOne()
+- [x] [PQ] client.publication.fetchAll(request)
+- [x] [Q] client.publication.validateMetadata(request)
+- [x] [PQ] client.publication.whoCollected({ publicationId })
+- [x] [PQ] client.publication(s).forSale({ profileId })
+- [x] [Q] client.publication.metadataStatus(request)
```
Comment (krzysu) [inline packages/client/TODO.md:101]: notice the differences between what I suggested before

```
- [Q] client.publication.fetch(request)
- [PQ] client.publication.fetchAll(request)
- [Q] client.publication.validateMetadata(request)
- [PQ] client.publication.whoCollected({ publicationId })
- [PQ] client.publication.allForSale({ profileId })
- [Q] client.publication.metadataStatus(request)
```

all paginated queries could also use the pagination helper as it was merged already

there are two methods that I'd rename to make them more meaningful
- client.publication.**allWallets**WhoCollected (this one I missed before)
- client.publication.**all**ForSale

**Review by krzysu — APPROVED** [review]
some small renaming and use of pagination helper are missing, the rest looks good!
merge when you are ready 😄

**Pull request merged**